### PR TITLE
More build output changes

### PR DESCRIPTION
**Pull request by jaredpar**
Changes:

- Remove ProducesNoOutput usage as it's no longer needed.  In every case Settings.props or Versions.props can be imported directly
- Remove Dependencies.sln as it's no longer used.
- Remove unnecessary extra files in MicroBuild.

### CODE REVIEW

File: build/Targets/Versions.props [hunk 1]
```diff
@@ -41,6 +41,10 @@
     <!-- NuGet version -->
     <NuGetReleaseVersion>$(RoslynFileVersionBase)</NuGetReleaseVersion>
     <NuGetPerBuildPreReleaseVersion>$(NuGetPreReleaseVersion)-$(BuildNumberFiveDigitDateStamp)-$(BuildNumberBuildOfTheDayPadded)</NuGetPerBuildPreReleaseVersion>
+    <NuGetPackageRoot Condition="'$(NuGetPackageRoot)' == ''">$(NUGET_PACKAGES)</NuGetPackageRoot> <!-- Respect environment variable if set -->
```
Comment (jasonmalinowski) [inline build/Targets/Versions.props]: Could you add a comment to the file explaining why it's in Versions.props? This seems like it really should be in Settings.props where it was.

File: build/config/RepoUtilData.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
                 "Microsoft.CodeAnalysis.Analyzers",
                 "Microsoft.VisualBasic",
                 "Microsoft.Composition",
+                "MicroBuild.*",
```
Comment (jasonmalinowski) [inline build/config/RepoUtilData.json]: Where is it documented what this does?

File: build/Targets/Versions.props (hunk 1 above)
Comment (jaredpar) [inline build/Targets/Versions.props]: A considerable portion of the values defined in Version.props, and by extension Dependencies.props, are NuGet packages.  Not having the root for which to find them reduced the values of these files.  

Importing Settings.props is a bit of a heavier hand is it brings along a lot of properties which are specific to building.  There are a number of build files that are just tooling that can't always import these values.  Hence I broke it up this way. 

I've tried to document the intent of these files in the following README.md.  Happy to update if you feel necessary

https://github.com/dotnet/roslyn/blob/master/build/Targets/README.md

File: build/config/RepoUtilData.json (hunk 2 above)
Comment (jaredpar) [inline build/config/RepoUtilData.json]: Latest commit on this PR updated the README.md for RepoUtil to describe this section.

File: build/Targets/Versions.props (hunk 1 above)
Comment (tmat) [inline build/Targets/Versions.props]: Perhaps move nuget related properties to a separate NuGet.props like so:
https://github.com/dotnet/interactive-window/blob/master/build/Toolset/NuGet.props
https://github.com/dotnet/symreader-portable/blob/master/build/Toolset/NuGet.props
https://github.com/dotnet/symreader/blob/master/build/Toolset/NuGet.props
https://github.com/dotnet/symstore/blob/master/build/Toolset/NuGet.props

**Pull request merged**